nibabel/filebasedimages.py

```diff
@@ -347,7 +356,16 @@ def to_filename(self, filename):
         -------
         None
         '''
-        self.file_map = self.filespec_to_file_map(filename)
+        self.file_map = file_map = self.filespec_to_file_map(filename)
+        remove_env = os.environ.get('NIBABEL_REMOVE_BEFORE_TO_FILENAME', None)
+        for _, fh in file_map.items():
+            if not isinstance(fh, FileHolder):
+                continue
+            if os.path.lexists(fh.filename) and (remove_env or _is_annex_symlink(fh.filename)):
+                # Remove previous file where new file would be saved
+                # Necessary e.g. for cases where file is a symlink pointing
+                # to some non-writable file (e.g. under git annex control)
+                os.unlink(fh.filename)
         self.to_file_map()
```

Comment:
or @matthew-brett do you think it would remain kosher to shift this functionality even deeper into to_file_map? ;)